practitioner-service/src/test/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidationTests.java

```diff
@@ -1,20 +1,41 @@
 package uk.ac.ox.ndph.mts.practitioner_service.validation;
 
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.converter.ConvertWith;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.reactive.function.client.WebClient;
 import uk.ac.ox.ndph.mts.practitioner_service.NullableConverter;
+import uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClient;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.RestException;
 import uk.ac.ox.ndph.mts.practitioner_service.model.RoleAssignment;
+
+import java.net.HttpURLConnection;
+
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClientTest.queueErrorResponse;
+import static uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClientTest.queueRoleResponse;
 
 @ExtendWith(MockitoExtension.class)
 class RoleAssignmentValidationTests {
 
+    private MockWebServer mockBackEnd;
+
+    private RoleAssignmentValidation validator;
+
+    @BeforeEach void initEach() throws Exception {
+        this.mockBackEnd = new MockWebServer();
+        this.mockBackEnd.start();
+        this.validator = new RoleAssignmentValidation(new WebFluxRoleServiceClient(WebClient.builder(), String.format("http://localhost:%s", mockBackEnd.getPort())));
```

Comment:
I think the validator should get a mock of the WebFluxRoleServiceClient. Then on the tests below you can return true/false by using something like:
```java
    @Mock
    private WebFluxRoleServiceClient roleServiceClient;

    @Captor
    ArgumentCaptor<String> roleIdCaptor;

...

    @Test
    void TestRoleAssignmentValidation_WhenInvalidRole_ReturnsInvalidResponse() throws Exception {
        // Arrange
        when(roleServiceClient.roleIdExists(anyString())).thenReturn(false);
        String roleId = "roleId";
        RoleAssignment roleAssignment = new RoleAssignment("practitionerId", "siteId", roleId);
        // Act
        var result = validator.validate(roleAssignment);
        // Assert
        Mockito.verify(roleServiceClient).roleIdExists(roleIdCaptor.capture());
        var value = roleIdCaptor.getValue();
        assertThat(result.isValid(), is(false));
        assertThat(roleId, equalTo(value));
        assertThat(result.getErrorMessage(), containsString("roleId"));
    }
```

Reply:
OK will do